src/App.jsx

```diff
@@ -14,33 +15,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case 'alphabetically':
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case 'length':
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const handleSort = newSortField => {
+    setSortField(newSortField);
+  };
```

Comment:
Consider checking if the `newSortField` is different from the current `sortField` before calling `setSortField`. This will prevent unnecessary re-renders when the same sort field is selected multiple times.